dexcom/fetch/runner.go

```diff
@@ -124,17 +124,16 @@ func (r *Runner) Run(ctx context.Context, tsk *task.Task) bool {
 	}
 
 	if !skipToAvoidDexcomBackup {
-		tsk.ClearError()
+		ResetTask(tsk)
 
 		if serverSessionToken, sErr := r.AuthClient().ServerSessionToken(); sErr != nil {
-			tsk.AppendError(errors.Wrap(sErr, "unable to get server session token"))
+			ErrorOrRetryTask(tsk, errors.Wrap(sErr, "unable to get server session token"))
 		} else {
 			ctx = auth.NewContextWithServerSessionToken(ctx, serverSessionToken)
-
 			if taskRunner, tErr := NewTaskRunner(r, tsk); tErr != nil {
-				tsk.AppendError(errors.Wrap(tErr, "unable to create task runner"))
+				ErrorOrRetryTask(tsk, errors.Wrap(tErr, "unable to create task runner"))
```

Comment:
If this block is executed, this means that there is bug in the code. It's better not to fail the task here, but keep on retrying (until the bug is eventually fixed).